test/document_storage_tests/documents_tests/actions/insert_documents.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2023 Codenotary Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package actions
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/gavv/httpexpect/v2"
+)
+
+func InsertOneDocumentWithMultipleFields(expect *httpexpect.Expect, sessionID string, collection *httpexpect.Object, document map[string]interface{}) *httpexpect.Object {
+	collectionName := collection.Value("collection").Object().Value("name").String().Raw()
+
+	payload := map[string]interface{}{
+		"documents": []interface{}{
+			document,
+		},
+	}
+
+	keys := reflect.ValueOf(document).MapKeys()
+	field := keys[1].String()
+	value := document[keys[1].String()]
```

Comment:
You can get Keys and values by simple iteration, without using reflect as far as I see

Reply:
Not a problem, but I would say that is not the usual way,  Reflection package is useful when access to the data is restricted or in non strict types...

Reply:
Ok. I'll change it then. Thanks.

Reply:
there is no guarantee on order (also in reflect as you have map). Just pass to the function with which field/value you want to have search later.